client/src/components/graph-line.jsx

```diff
@@ -0,0 +1,11 @@
+import { memo } from 'preact/compat'
```

Comment:
Given that this component is very tightly coupled to Graph (and takes input generated by a function from that component), it would make more sense to define this in `graph.js`, especially since it is not reusable